src/App.tsx

```diff
@@ -426,25 +446,58 @@ function App() {
         if (!clientSimulator) {
             return false;
         }
-        const productConcentration =
-            clientSimulator.getCurrentConcentrations(productName)[productName];
-        const reactantConcentration = inputConcentration[ADJUSTABLE_AGENT] || 0;
 
         if (!clientSimulator.isMixed()) {
             setEquilibriumFeedbackTimeout("Not yet!");
             return false;
         }
+        // this will always be defined for the current run, but since there are
+        // different agents in each module, typescript fears it will be undefined
+        const currentInputConcentration = inputConcentration[ADJUSTABLE_AGENT];
+        if (currentInputConcentration === undefined) {
+            return false;
+        }
+        const concentrations =
+            clientSimulator.getCurrentConcentrations(productName);
+        const productConcentration = concentrations[productName];
+        const reactantConcentration = concentrations[ADJUSTABLE_AGENT];
+
+        const currentTime = indexToTime(
+            currentProductConcentrationArray.length,
+            timeFactor,
+            simulationData.timeUnit
+        );
         const { newArray, index } = insertValueSorted(
-            inputEquilibriumConcentrations,
+            recordedReactantConcentrations,
             reactantConcentration
         );
-        setInputEquilibriumConcentrations(newArray as number[]);
-        const newProductArray = insertIntoArray(
+        setRecordedReactantConcentration(newArray as number[]);
+        updateArrayInState(
             productEquilibriumConcentrations,
             index,
-            productConcentration
+            productConcentration,
+            setProductEquilibriumConcentrations
         );
-        setProductEquilibriumConcentrations(newProductArray as number[]);
+        updateArrayInState(
+            recordedInputConcentration,
+            index,
+            currentInputConcentration,
+            setRecordedInputConcentration
+        );
+        updateArrayInState(
+            timeToReachEquilibrium,
+            index,
+            currentTime,
+            setTimeToReachEquilibrium
+        );
```

Comment:
Not totally sure this function is a huge benefit over just inlining

```
const newEquilibriumArray = insertIntoArray(
    timeToReachEquillibrium,
    index,
    currentTime
);
setTimeToReachEquilibrium(newEquilibriumArray);
```

Reply:
yeah that's what I had before, but since there are a bunch of them, it just felt nicer to have them each done all at once